### PR TITLE
Added http code enums, and expanded http code support in vcWebFile.cpp

### PR DESCRIPTION
**Pull request by int-Frank**
[AB#187](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/187)

### CODE REVIEW

File: src/vcHTTP.cpp [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+  case vcHTTP_ServerError:
+    return "Internal Server Error";
+  case vcHTTP_NotImplemented:
+    return "Not Implemented";
+  case vcHTTP_ServiceUnavailable:
+    return "Service Unavailable";
+  case vcHTTP_BadHTTPVersion:
+    return "HTTP Version not supported";
+  default:
+    return "Internal Server Error";
```
Comment (samsinsane) [inline src/vcHTTP.cpp]: We shouldn't have a `default` (new entries in the `vcHTTP` enum should be handled correctly) and the fallback string shouldn't be an internal server error - only the 5xx range are server errors. A better fallback would be "unknown HTTP status code" or something.

File: src/vcHTTP.cpp [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    vcHTTP_SplitAndInject(&pHeaders->cookies, pCookieStr, ';', '=', true);
+
+    udFree(pCookieStr);
+  }
+
+  if (!pHeaders->tokens.Get("Cookie").IsObject())
+    pHeaders->tokens.Set("Cookie = {}");
+}
+
+const char* vcHTTP_StatusAsText(vcHTTPStatus status)
```
Comment (samsinsane) [inline src/vcHTTP.cpp]: @OsiJr This function isn't used and I've made two comments against it, should vcHTTP contain so much unused functionality?

**Comment by int-Frank** [comment]
@samsinsane Righto

**Comment by OsiJr** [comment]
Closing this PR due to decay. It should be rebased and refreshed before reopening.

**Pull request closed without merging**